R/calc_aggregate_counts.R

```diff
@@ -60,17 +88,45 @@ calc_aggregate_counts <- function(
             collapse_vaccine,
             mutate(., Residents.Vadmin = ifelse(
                 is.na(.$Residents.Vadmin), .$Residents.Completed, .$Residents.Vadmin)),
-            .) %>%
-        tidyr::pivot_longer(
-            -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
-        filter(!is.na(UCLA)) %>%
-        group_by(State, Measure) %>%
-        mutate(has_statewide = "STATEWIDE" %in% Name) %>%
-        # if state wide and other counts exist for a measure only use statewide
-        filter(!(has_statewide & Name != "STATEWIDE")) %>%
-        group_by(State, Measure) %>%
-        summarise(UCLA = sum_na_rm(UCLA), .groups = "drop") %>%
-        full_join(mp_data, by = c("State", "Measure")) %>%
+            .)
+
+    if(all_dates){
+        state_df <- state_wide_df %>%
+            mutate(Date = lubridate::floor_date(Date, round_)) %>%
+            tidyr::pivot_longer(
+                -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
+            filter(!is.na(UCLA)) %>%
+            group_by(State, Date, Measure, Name) %>%
+            summarize(UCLA = max_na_rm(UCLA), .groups = "drop_last") %>%
+            mutate(has_statewide = "STATEWIDE" %in% Name) %>%
+            # if state wide and other counts exist for a measure only use statewide
+            filter(!(has_statewide & Name != "STATEWIDE")) %>%
+            group_by(State, Date, Measure) %>%
+            summarise(UCLA = sum_na_rm(UCLA), .groups = "drop")
+
+        comb_df <- state_df %>%
+            full_join(mp_data, by = c("State", "Measure", "Date"))
+    }
+    else{
+        state_df <- state_wide_df %>%
+            tidyr::pivot_longer(
+                -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
+            filter(!is.na(UCLA)) %>%
+            group_by(State, Measure) %>%
+            mutate(has_statewide = "STATEWIDE" %in% Name) %>%
```

Comment:
I would add another condition to this, if "STATE-WIDE" is %in% Name. Suggesting this because we leave dashes in when name-cleaning, and if we don't have a clean spelling yet for an instance of "STATE-WIDE", this won't detect it. Hoping the "STATE-WIDE" condition will never be met, but it might??

Reply:
good point will do